### PR TITLE
Adding createClassResolver helper to react-compose

### PR DESCRIPTION
**Pull request by dzearing**
Separating a PR out for the class resolver utility. Expected usage would be:

```tsx
import * as classes from './some.module.css';

compose(Thing, {
  classes: createClassResolver(classes)
});
```

### CODE REVIEW

File: packages/react-compose/src/createClassResolver.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  it('can resolve enums', () => {
+    // Can resolve
+    expect(classResolver({ size: 'small' }, slots)).toEqual({
+      root: 'root small',
+      slot1: 'slot1',
+      slot2: 'slot2',
+    });
+  });
+
+  it('can resolve mixed content, including a className in props', () => {
```
Comment (xugao) [inline packages/react-compose/src/createClassResolver.test.ts:39]: lets add test for when `state` has properties that are not mapped to classes:
```
classResolver({ className: 'foo', primary: true, size: 'medium', bar: 'bar' }, slots)
```

File: packages/react-compose/src/createClassResolver.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    expect(classResolver({}, slots)).toEqual({
+      root: 'root',
+      slot1: 'slot1',
+      slot2: 'slot2',
+    });
+  });
+
+  it('can resolve modifiers', () => {
+    expect(classResolver({ primary: true }, slots)).toEqual({
+      root: 'root primary',
```
Comment (xugao) [inline packages/react-compose/src/createClassResolver.test.ts:24]: with our scss conversions, we are actually adding modifier classes onto every slot. that way the scss can be written in a cleaner way.

Comment (dzearing) [reply packages/react-compose/src/createClassResolver.test.ts:24]: changed.

File: packages/react-compose/src/createClassResolver.test.ts (hunk 1 above)
Comment (dzearing) [reply packages/react-compose/src/createClassResolver.test.ts:46]: updated.

**Pull request merged**